jobsdb/jobsdb.go

```diff
@@ -665,10 +682,18 @@ func (jd *HandleT) addNewDS(appendLast bool, insertBeforeDS dataSetT) dataSetT {
 		jd.assert(len(dList) > 0, fmt.Sprintf("len(dList): %d <= 0", len(dList)))
 		for idx, ds := range dList {
 			if ds.Index == insertBeforeDS.Index {
-				//We never insert before the first element
-				jd.assert(idx > 0, fmt.Sprintf("idx: %d <= 0", idx))
+				// We never insert before the first element
+				// We do now.
+				//TODO: Review this carefully
```

Comment:
@chandumlg

Reply:
@soumyadebm @SumanthPuram This change is regarding ds numbering. Please review this